actionners/kubernetes/sysdig/sysdig.go

```diff
@@ -0,0 +1,292 @@
+)
+
+type Parameters struct {
+	Image      string `mapstructure:"image"`
+	Scope      string `mapstructure:"scope" validate:"oneof=pod node"`
+	Duration   int    `mapstructure:"duration" validate:"gte=0,lte=30"`
+	BufferSize int    `mapstructure:"buffer_size" validate:"gte=128"`
+}
+
+const (
+	baseName     string = "falco-talon-sysdig-"
+	defaultImage string = "issif/sysdig:latest"
+	// defaultImage       string = "sysdig/sysdig:latest"
+	defaultScope       string = "pod"
+	defaultTTL         int    = 60
+	defaultDuration    int    = 5
+	defaultMaxDuration int    = 30
+	defaultBufferSize  int    = 2048
+)
+
+type Actionner struct{}
+
+func Register() *Actionner {
+	return new(Actionner)
+}
+
+func (a Actionner) Init() error {
+	return k8s.Init()
+}
+
+func (a Actionner) Information() models.Information {
+	return models.Information{
+		Name:                 Name,
+		FullName:             Category + ":" + Name,
+		Category:             Category,
+		Description:          Description,
+		Source:               Source,
+		RequiredOutputFields: RequiredOutputFields,
+		Permissions:          Permissions,
+		Example:              Example,
+		Continue:             Continue,
+		AllowOutput:          AllowOutput,
+		RequireOutput:        RequireOutput,
+	}
+}
+
+func (a Actionner) Parameters() models.Parameters {
+	return Parameters{
+		Duration:   defaultDuration,
+		Scope:      defaultScope,
+		BufferSize: defaultBufferSize,
+		Image:      defaultImage,
+	}
+}
+
+func (a Actionner) Checks(event *events.Event, _ *rules.Action) error {
+	if event.Source == "k8s_audit" {
+		event.OutputFields["k8s.ns.name"] = event.OutputFields["ka.target.namespace"]
+		if event.OutputFields["ka.target.pod.name"] != "" {
+			event.OutputFields["k8s.pod.name"] = event.OutputFields["ka.target.pod.name"]
+		} else {
+			event.OutputFields["k8s.pod.name"] = event.OutputFields["ka.target.name"]
+		}
+	}
+	return k8sChecks.CheckPodExist(event)
+}
+
+func (a Actionner) Run(event *events.Event, action *rules.Action) (utils.LogLine, *models.Data, error) {
+	podName := event.GetPodName()
+	namespace := event.GetNamespaceName()
+
+	objects := map[string]string{
+		"pod":       podName,
+		"namespace": namespace,
+	}
+
+	var parameters Parameters
+	err := utils.DecodeParams(action.GetParameters(), &parameters)
+	if err != nil {
+		return utils.LogLine{
+			Objects: nil,
+			Error:   err.Error(),
+			Status:  utils.FailureStr,
+		}, nil, err
+	}
+
+	if parameters.Duration == 0 {
+		parameters.Duration = defaultDuration
+	}
+
+	if parameters.Duration > 30 {
+		parameters.Duration = defaultMaxDuration
+	}
+
+	if parameters.Image == "" {
+		parameters.Image = defaultImage
+	}
+
+	if parameters.Scope == "" {
+		parameters.Scope = defaultScope
+	}
+
+	client := k8s.GetClient()
+
+	pod, _ := client.GetPod(podName, namespace)
+	containers := k8s.GetContainers(pod)
+	if len(containers) == 0 {
```

Comment:
Is it possible to have a pod with no containers?

Reply:
normally not, i think i added it to please the linting

Reply:
ok